File: unity/Assets/Scripts/Multiplayer/Matchmaking/MatchmakingHeartbeatResponse.cs
using Michitai.Multiplayer.Errors;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Matchmaking
{
    /// <summary>
    /// Response returned when a matchmaking heartbeat is successfully sent.
    /// Used to maintain the player's presence in the lobby.
    /// </summary>
    [System.Serializable]
    public class MatchmakingHeartbeatResponse : ApiResponse<EMatchmakingHeartbeatError>
    {
        /// <summary>
        /// Status message indicating the heartbeat was received.
        /// </summary>
        public string status;
    }
}

File: java/src/main/java/com/michitai/multiplayer/matchmaking/MatchmakingHeartbeatResponse.java
package com.michitai.multiplayer.matchmaking;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.ApiResponse;

/**
 * Response returned when a matchmaking heartbeat is successfully sent.
 */
public class MatchmakingHeartbeatResponse extends ApiResponse {
    @JsonProperty("status")
    private String status;

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }
}
